src/Geocoder.php

```diff
@@ -8,216 +8,252 @@
+    $param_array = array();
+    foreach ($parameters as $key => $value) {
+      $param_array[] = "$key=" . urlencode($value);
+    }
+    $param_array[] = "$key=" . urlencode($value);
 
-    static function focus($latitude, $longitude)
-      {
-      return array("focus" => "$latitude,$longitude");
-      }
+    // glue the array together unto a string with elements connected with '&'
+    $params = implode("&", $param_array);
 
-    static function numberFocusResults($number_focus_results)
-      {
-      return array("n-focus-results" => "$number_focus_results");
-      }
-      
-      static function inputType($input_type)
-      {
-        return array("input-type" => "$input_type");
-      }
-      
-      static function preferLand($land)
-      {
-        if ($land)
-          return array("prefer-land" => "true");
-        else
-          return array("prefer-land" => "false");
-      }
-      
-    static function clipToCountry($country)
-      {
-      return array("clip-to-country" => "$country");
-      }
+    // put the whole URL together now
+    $url = $this->baseUrl . $command . "?" . $params;
 
-    static function clipToCircle($latitude, $longitude, $radius)
-      {
-      return array("clip-to-circle" => "$latitude,$longitude,$radius");
-      }
+    // cal the server
+    $json = $this->call($url);
 
-    static function clipToBoundingBox($south_lat, $west_lng, $north_lat, $east_lng)
-      {
-      return array("clip-to-bounding-box" => "$south_lat,$west_lng,$north_lat,$east_lng");
-      }
+    $data = json_decode($json, true);
 
-    static function clipToPolygon($points=array())
-      {
-      return array("clip-to-polygon" => implode(",", $points));
-      }
+    if (isset($data["error"])) {
+      $this->error["code"] = $data["error"]["code"];
+      $this->error["message"] = $data["error"]["message"];
+      $data = false;
+    }
 
+    return $data;
+  }
+
+
+  // Make the call to the API server
+  private function call($url)
+  {
+    $handle = curl_init();
+
+    // set the options
+    curl_setopt($handle, CURLOPT_URL, $url);
+    curl_setopt($handle, CURLOPT_HTTPHEADER, array("X-W3W-Wrapper: " . $this->header));
+    curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($handle, CURLOPT_ENCODING, "");
+    curl_setopt($handle, CURLOPT_MAXREDIRS, 10);
+    curl_setopt($handle, CURLOPT_TIMEOUT, 30);
+    curl_setopt($handle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
+    curl_setopt($handle, CURLOPT_CUSTOMREQUEST, "GET");
+
+    // make the call
+    $output = curl_exec($handle);
+    if (!$output) {
+      $this->error["code"] = "BadConnection";
+      $this->error["message"] = curl_error($handle);
     }
 
+    curl_close($handle);
+
+    // return the json
+    return $output;
+  }
+
+}
+
+
+// These are static funcntions that make options to be passed into autosuggest
```

Comment:
```suggestion
// These are static functions that return options to be passed into autosuggest
```

Reply:
Technically this still creates/make options but I guess the message still not quite relayed correctly.